// @mui/x-data-grid-generator/node/renderer/renderIncoterm.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.renderIncoterm = renderIncoterm;
var React = _interopRequireWildcard(require("react"));
var _Tooltip = _interopRequireDefault(require("@mui/material/Tooltip"));
var _Box = _interopRequireDefault(require("@mui/material/Box"));
var _Info = _interopRequireDefault(require("@mui/icons-material/Info"));
var _jsxRuntime = require("react/jsx-runtime");
const Incoterm = /*#__PURE__*/React.memo(function Incoterm(props) {
  const {
    value
  } = props;
  if (!value) {
    return null;
  }
  const valueStr = value.toString();
  const tooltip = valueStr.slice(valueStr.indexOf('(') + 1, valueStr.indexOf(')'));
  const code = valueStr.slice(0, valueStr.indexOf('(')).trim();
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_Box.default, {
    sx: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between'
    },
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("span", {
      children: code
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_Tooltip.default, {
      title: tooltip,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_Info.default, {
        sx: {
          color: '#2196f3',
          alignSelf: 'center',
          ml: '8px'
        }
      })
    })]
  });
});
function renderIncoterm(params) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(Incoterm, {
    value: params.value
  });
}
