// @mui/x-data-grid-generator/node/renderer/renderLink.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DemoLink = void 0;
exports.renderLink = renderLink;
var React = _interopRequireWildcard(require("react"));
var _styles = require("@mui/material/styles");
var _jsxRuntime = require("react/jsx-runtime");
const Link = (0, _styles.styled)('a')({
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
  overflow: 'hidden',
  color: 'inherit'
});
const DemoLink = exports.DemoLink = /*#__PURE__*/React.memo(function DemoLink(props) {
  const handleClick = event => {
    event.preventDefault();
    event.stopPropagation();
  };
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(Link, {
    tabIndex: props.tabIndex,
    onClick: handleClick,
    href: props.href,
    children: props.children
  });
});
function renderLink(params) {
  if (params.value == null) {
    return '';
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(DemoLink, {
    href: params.value,
    tabIndex: params.tabIndex,
    children: params.value
  });
}
